core/src/main/java/org/jclouds/internal/BaseView.java

```diff
@@ -45,7 +46,7 @@ protected BaseView(@Provider Context backend, @Provider TypeToken<? extends Cont
    @SuppressWarnings("unchecked")
    @Override
    public <C extends Context> C unwrap(TypeToken<C> type) {
-      checkArgument(checkNotNull(type, "type").isAssignableFrom(backendType), "backend type: %s not assignable to %s", backendType, type);
+      checkArgument(TypeTokenUtils.isSupertypeOf(checkNotNull(type, "type"), backendType), "backend type: %s is not a supertype of %s", backendType, type);
```

Comment:
Should the error message be the other way around? I.e. `"type %s is not a supertype of backend type %s", type, backendType'?